sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -647,6 +653,46 @@ func (n *ProcessSizedElementsAndRestrictions) Split(f float64) ([]*FullValue, []
 	return p, r, nil
 }
 
+// Checkpoint splits the remaining work in a restriction into residuals to be resumed
+// later by the runner. This is done iff the underlying Splittable DoFn returns a resuming
+// ProcessContinuation. If the split occurs and the primary restriction is marked as done
+// my the RTracker, the Checkpoint fails as this is a potential data-loss case.
+func (n *ProcessSizedElementsAndRestrictions) Checkpoint() ([]*FullValue, error) {
+	// Get the watermark state immediately so that we don't overestimate our current watermark.
+	var pWeState interface{}
+	var rWeState interface{}
+	rWeState = n.wesInv.Invoke(n.PDo.we)
+	pWeState = rWeState
+	// If we've processed elements, the initial watermark estimator state will be set.
+	// In that case we should hold the output watermark at that initial state so that we don't
+	// Advance past where the current elements are holding the watermark
+	if n.initWeS != nil {
+		pWeState = n.initWeS
+	}
+	addContext := func(err error) error {
+		return errors.WithContext(err, "Attempting checkpoint in ProcessSizedElementsAndRestrictions")
+	}
+
+	// Errors checking.
+	if n.rt == nil {
+		return nil, addContext(errors.New("Restriction tracker missing."))
+	}
+	if err := n.rt.GetError(); err != nil {
+		return nil, addContext(err)
+	}
+
+	_, r, err := n.singleWindowSplit(0.0, pWeState, rWeState)
+	if err != nil {
+		return nil, addContext(err)
+	}
```

Comment:
```suggestion
	_, r, err := n.Split(0.0)
```

I like moving this into its own function in sdf.go, but that doesn't mean we can't still leverage Split